apps/rush-lib/src/utilities/Utilities.ts

```diff
@@ -564,6 +564,65 @@ export class Utilities {
     return new Error('Unable to find rush.json configuration file');
   }
 
+  /**
+    * Gets a path to a file or folder on disk with the exact character casing that exists on disk.
+    *
+    * If the file or folder specified doesn't exist on disk, undefined is returned
+    *
+    * @example
+    * getCorrectlyCasedPath('c:\\uppercase-folder\\LOWERCASE-FILE') => 'C:\\UPPERCASE-FOLDER\\lowercase-file'
+    * getCorrectlyCasedPath('/uppercase-folder/LOWERCASE-FILE') => '/UPPERCASE-FOLDER/lowercase-file'
+    *
+    * @remarks
+    * This function assumes the path passed to it is absolute and is on the local filesystem
+    */
+  public static getCorrectlyCasedPath(pathToNormalize: string): string | undefined {
```

Comment:
Take a look at [true-case-path](https://www.npmjs.com/package/true-case-path)

Reply:
I like that. Refactoring to use that.